### PR TITLE
bluetooth: mesh: setting iv update sequence limit by kconfig

### PR DESCRIPTION
**Pull request by haoyanbiao**
Signed-off-by: YanBiao Hao <haoyanbiao@xiaomi.com>

### CODE REVIEW

File: subsys/bluetooth/mesh/Kconfig [hunk 1]
```diff
@@ -59,6 +59,13 @@ config BT_MESH_PROVISIONER
 	help
 	  Enable this option to have support for provisioning remote devices.
 
+config BT_MESH_IV_UPDATE_SEQ_LIMIT
+	hex "the sequence limit to start iv update"
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/Kconfig:63]:
```suggestion
	hex "Sequence number limit to start iv update"
```

File: subsys/bluetooth/mesh/Kconfig [hunk 2]
```diff
@@ -59,6 +59,13 @@ config BT_MESH_PROVISIONER
...
+	default 0x7A1200
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/Kconfig:64]: Is there any reason for changing the default from the original `0x800000`?

**Review by trond-snekvik — CHANGES_REQUESTED** [review]
Nice addition!

Could you please move this entry to go in right above `config BT_MESH_IVU_DIVIDER` instead? This file isn't very well ordered, but it'd be nice to keep similar entries close to each other.

File: subsys/bluetooth/mesh/Kconfig (hunk 2 above)
Comment (haoyanbiao) [reply subsys/bluetooth/mesh/Kconfig:64]: I mistook it for decimal and changed it back to the default value 0x8000000

File: subsys/bluetooth/mesh/Kconfig (hunk 1 above)
Comment (haoyanbiao) [reply subsys/bluetooth/mesh/Kconfig]: Ok

File: subsys/bluetooth/mesh/Kconfig [hunk 3]
```diff
@@ -267,6 +267,13 @@ config BT_MESH_ADV_STACK_SIZE
 	  NOTE: This is an advanced setting and should not be changed unless
 	  absolutely necessary
 
+config BT_MESH_IV_UPDATE_SEQ_LIMIT
+	hex "the Sequence number limit to start iv update"
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/Kconfig:271]:
```suggestion
	hex "Sequence number limit to start iv update"
```

**Review by trond-snekvik — APPROVED** [review]

**Review by LingaoM — APPROVED** [review]

File: subsys/bluetooth/mesh/Kconfig (hunk 2 above)
Comment (jhedberg) [reply subsys/bluetooth/mesh/Kconfig]: I'm confused. Wasn't the original `8000000` in decimal?

File: subsys/bluetooth/mesh/net.c [hunk 4]
```diff
@@ -44,7 +44,7 @@
 #define LOOPBACK_BUF_SUB(buf) (*(struct bt_mesh_subnet **)net_buf_user_data(buf))
 
 /* Seq limit after IV Update is triggered */
-#define IV_UPDATE_SEQ_LIMIT 8000000
```
Comment (jhedberg) [inline subsys/bluetooth/mesh/net.c:47]: I saw the comment from @trond-snekvik mentioning the original being `0x8000000` but unless I'm missing something that's not the case?

Comment (LingaoM) [reply subsys/bluetooth/mesh/net.c:47]: I think it is indeed decimal, in fact there is not much difference, `previous hex(8000000)=0x7A1200` vs `current 0x800000`.

Comment (haoyanbiao) [reply subsys/bluetooth/mesh/net.c:47]: This value will be changed according to the needs of the appliance, so the initial value has no special significance.

Comment (jhedberg) [reply subsys/bluetooth/mesh/net.c:47]: Yes, I agree.

**Review by jhedberg — APPROVED** [review]

**Pull request merged**